src/facade/dragonfly_connection.cc

```diff
@@ -1387,11 +1418,11 @@ unsigned Connection::WeakRef::Thread() const {
 }
 
 Connection* Connection::WeakRef::Get() const {
-  DCHECK_EQ(ProactorBase::me()->GetPoolIndex(), int(thread_));
-  // The connection can only be deleted on this thread, so
-  // this pointer is valid until the next suspension.
-  // Note: keeping a shared_ptr doesn't prolong the lifetime because
-  // it doesn't manage the underlying connection. See definition of `self_`.
+  // DCHECK_EQ(ProactorBase::me()->GetPoolIndex(), int(thread_));
```

Comment:
This dcheck is here because using the pointer from Get() in another thread is unsafe

Reply:
i see, let me change the use of Get() as you suggested below...

Reply:
now I have improved the notification logic, this DCHECK has now come back :-)